lib/buffer.js

```diff
@@ -61,8 +61,8 @@ export default class Buffer {
       this.markers[message.key] = marker
       this.messages.add(message)
       this.applyMarker(message, this.editors)
-      marker.onDidChange(({ newHeadBufferPosition, newTailBufferPosition, isValid }) => {
-        if (isValid) {
+      marker.onDidChange(({ oldHeadBufferPosition, newHeadBufferPosition, newTailBufferPosition, isValid }) => {
```

Comment:
Shouldn't there be a related change to whatever is calling this so the `oldHeadBufferPosition` is passed in?